Paraswap-Agents/AdminOperations/src/agent.ts

```diff
@@ -0,0 +1,62 @@
+import { BigNumber } from "ethers";
+import { Finding, HandleTransaction, TransactionEvent, LogDescription, getEthersProvider } from "forta-agent";
+import { createEventFinding, createFunctionFinding } from "./findings";
+import { ADMIN_OPERATIONS, AUGUSTUS_SWAPPER_CONTRACTS } from "./utils";
+
+let augustusSwapperContract = "";
+
+const initialize = async () => {
+  // set the contract address based on the networkId.
+  const provider = getEthersProvider();
+  const network = await provider.getNetwork();
+  augustusSwapperContract = AUGUSTUS_SWAPPER_CONTRACTS[network.chainId];
+};
+
+export const provideHandleTransaction =
+  (_augustusSwapperContract?: string): HandleTransaction =>
+  async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    if (!_augustusSwapperContract) _augustusSwapperContract = augustusSwapperContract;
+
+    // get event logs
+    const logs: LogDescription[] = txEvent.filterLog(
+      [ADMIN_OPERATIONS[5], ADMIN_OPERATIONS[6]],
+      _augustusSwapperContract
+    );
+    // get function calls
+    const functionsCalls = txEvent.filterFunction(
+      [ADMIN_OPERATIONS[0], ADMIN_OPERATIONS[2], ADMIN_OPERATIONS[3], ADMIN_OPERATIONS[4]],
+      _augustusSwapperContract
+    );
+    // generate findings for logs.
+    logs.forEach((log) => {
+      findings.push(createEventFinding(log));
+    });
+    // get Transfer event logs.
+    let transferLogs = txEvent.filterLog([ADMIN_OPERATIONS[1]], _augustusSwapperContract);
+
+    functionsCalls.forEach((call) => {
+      // for transferTokens call, generate findings only if the Transfer event was emitted.
+      if (call.name == "transferTokens") {
+        for (let transfer of transferLogs) {
+          if (
+            transfer.args.from.toLowerCase() == txEvent.from.toLowerCase() &&
+            transfer.args.to.toLowerCase() == call.args.destination.toLowerCase() &&
+            BigNumber.from(transfer.args.value).eq(BigNumber.from(call.args.amount))
+          ) {
+            findings.push(createFunctionFinding(call));
+            transferLogs.splice(transferLogs.indexOf(transfer), 1);
+            break;
+          }
+        }
+        // generate findings for the other function calls
+      } else findings.push(createFunctionFinding(call));
```

Comment:
As it currently is, we are only creating alerts when the `transferTokens` function is called **AND** the `Transfer` event is emitted.

However, in the `transferTokens` function, there is an [`if` statement](https://etherscan.io/address/0xDEF171Fe48CF0115B1d80b88dc8eAB59176FEe57#code#F1#L1610) that checks whether it is an ETH transfer or an ERC-20 transfer.

Because of this `if` statement, we need to also create alerts when there is an ETH transfer, which is when the `transferTokens` function is called **but does not** emit a `Transfer` event.

Reply:
Oh I have missed that ETH transfer part using the low level `call` method. Nice catch 👍 
Agent updated so that all `transferTokens` calls generate findings. The `Transfer` event is no longer needed.